cpp/src/Slice/SliceUtil.cpp

```diff
@@ -788,3 +788,72 @@ bool Slice::opCompressReturn(const OperationPtr& op)
 {
     return opCompress(op, false);
 }
+
+string Slice::getDeprecateReason(const ContainedPtr& p, bool checkContainer)
```

Comment:
I wondering if we should use an scoped enum `CheckContainter {yes, no}` to make code more readable

```
getDeprecateReason(p, CheckContainer::no);

getDeprecateReason(p, false);
```

Reply:
To me personally, introducing a new enum just for this seems a little overkill. I think it's easy enough to just find the source (or header once we name the parameters in them) and see what the name is.

In an ideal world, we could use named parameters, but C++ doesn't natively support them. Maybe this is also overkill, but there are ways of emulating named parameters by overloading the '=' operator to get stuff like this:
```
using CheckContainer = NamedWrapperThing<bool>;
...
getDeprecateReason(p, CheckContainer = false);
```